### PR TITLE
Ladybird/Appkit: Port Tab window to swift

### PR DESCRIPTION
**Pull request by Peredurz**
This is a direct port of the Objective-C++ code to Swift 6. I have looked at how the other two swift files were ported and tried to do that with this as well.

### CODE REVIEW

File: Ladybird/AppKit/UI/Tab.swift [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+    var favicon: NSImage?
+    var searchPanel: SearchPanel
+    var inspectorController: InspectorController?
+
+    static var defaultFavicon: NSImage {
+        struct Static {
+            static let defaultFavicon: NSImage = {
+                let defaultFaviconPath = try! Core.Resource.load_from_uri("resource://icons/48x48/app-browser.png" as StaticString)
+                let nsDefaultFaviconPath = Ladybird.string_to_ns_string(defaultFaviconPath.filesystem_path()!)
```
Comment (ADKaster) [inline Ladybird/AppKit/UI/Tab.swift:38]: We should be able to re-use my extension to Swift.String from https://github.com/LadybirdBrowser/ladybird/pull/1175 to convert AK::String to  Swift.Strings.

File: Ladybird/AppKit/UI/Tab.swift [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    }
+
+    func onFindInPageResult(currentMatchIndex: Int, totalMatchCount: Int?) {
+        self.searchPanel.onFindInPageResult(currentMatchIndex: currentMatchIndex, totalMatchCount: totalMatchCount)
+    }
+
+    override var isVisible: Bool {
+        didSet {
+            (self.web_view as? LadybirdWebView)?.handleVisibility(isVisible)
+            super.isVisible = isVisible
```
Comment (ADKaster) [inline Ladybird/AppKit/UI/Tab.swift:297]: Do we need to set super.isVisible in didSet? I thought from reading the docs that didSet was called after `set` and as such the value would ... already be set. if that makes sense.

**Review by ADKaster — COMMENTED** [review]
My main concerns with this are:

- Are we setting the right patterns to be 'standard' swift? Or are we carrying over too many C++ patterns
- Does this handle the `weakSelf` type concerns with the Objective C++ code we have, where if we aren't careful we end up creating strong reference cycles and keeping Tabs/WebViews/Windows alive forever.

**Comment by ADKaster** [comment]
This doesn't seem to build with Xcode 16.1 Beta 1 in CI:

```
UI/Tab.swift:10:8: error: no such module 'Ladybird.Utilities'
  8 | import Foundation
  9 | import AppKit
 10 | import Ladybird.Utilities
    |        `- error: no such module 'Ladybird.Utilities'
```

File: Ladybird/AppKit/UI/Tab.swift [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+import Foundation
+import AppKit
+import Ladybird.Utilities
+import LibCore
+import LibGfx
+import LibURL
+import LibWebView
+import SwiftUI
+public class Tab: NSWindow,LadybirdWebViewObserver {
```
[the remark this answers is not in the document]
Comment (nico) [reply Ladybird/AppKit/UI/Tab.swift:16]: *set up

[the remark this answers is not in the document]
Comment (Peredurz) [reply Ladybird/AppKit/UI/Tab.swift:16]: I sadly don't have that command available to me in the cli, so I have to do it manually.

File: Ladybird/AppKit/UI/Tab.swift [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+    private let WINDOW_WIDTH: CGFloat = 1000
+    private let WINDOW_HEIGHT: CGFloat = 800
```
[the remark this answers is not in the document]
Comment (Peredurz) [reply Ladybird/AppKit/UI/Tab.swift:19]: No, the swift pattern should be lowerCamelCase [https://google.github.io/swift/#global-constants](Swift Style Guide), I mainly focused on copying what was used already in the other swift files. But I'll update it to be correct and will be following the style guide.

**Comment by Peredurz** [comment]
I can't seem to make the swift files being built, so for now I'm closing this PR.

**Pull request closed without merging**